### PR TITLE
[1.x] Merge branch `1.3` into `1.x`

### PR DESCRIPTION
**Pull request by peternied**
### Description
Updates the 1.x branch to include all changes that have gone directly into 1.3 bypassing this branch.  Needed to keep CI build clean in downstream 1.x branches that are following this practice.

### Related Issues
- Blocking https://github.com/opensearch-project/security/pull/4207

### Check List
- [ ] ~New functionality includes testing.~
  - [ ] ~All tests pass~
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [X] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/publish-maven-snapshots.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Publish snapshots to maven
```
Comment (reta) [inline .github/workflows/publish-maven-snapshots.yml:1]: I think this GA could be exluded. It should be in `main` only (parametrized with branch)

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
Thanks a lot @peternied !

File: .github/workflows/publish-maven-snapshots.yml (hunk 1 above)
Comment (peternied) [reply .github/workflows/publish-maven-snapshots.yml:1]: This was added to self publish to maven so downstream branches would get changes more quickly [1] - which is needed to for the security build to get unblocked, is there another publishing process?

- [1] #6394

Comment (reta) [reply .github/workflows/publish-maven-snapshots.yml:1]: Oh I see, it is a different GA (confused it with one we have)

File: .github/workflows/publish-maven-snapshots.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+on:
+  workflow_dispatch:
+  push:
+      branches:
+        - main
```
Comment (reta) [inline .github/workflows/publish-maven-snapshots.yml:7]:
```suggestion
        - 1.x
```

File: .github/workflows/publish-maven-snapshots.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      id-token: write
+      contents: write
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: adopt
+          java-version: 11
```
Comment (reta) [inline .github/workflows/publish-maven-snapshots.yml:25]:
```suggestion
          java-version: 8
```

File: .github/workflows/publish-maven-snapshots.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  build-and-publish-snapshots:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      contents: write
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
```
Comment (reta) [inline .github/workflows/publish-maven-snapshots.yml:21]:
```suggestion
      - name: Set up JDK 8
```

File: .github/workflows/publish-maven-snapshots.yml (hunk 2 above)
Comment (reta) [reply .github/workflows/publish-maven-snapshots.yml:7]: Scratch that please, we use JDK 11 to build

File: .github/workflows/publish-maven-snapshots.yml (hunk 3 above)
Comment (reta) [reply .github/workflows/publish-maven-snapshots.yml:25]: Scratch that please, we use JDK 11 to build

**Review by reta — APPROVED** [review]

File: .github/workflows/publish-maven-snapshots.yml (hunk 4 above)
Comment (reta) [reply .github/workflows/publish-maven-snapshots.yml:21]: Scratch that please, we use JDK 11 to build

File: .github/workflows/publish-maven-snapshots.yml (hunk 2 above)
Comment (peternied) [reply .github/workflows/publish-maven-snapshots.yml:7]: FYI - I didn't make any changes to this job, its merged directly from what is in 1.3.  Going to keep this unchanged

File: .github/workflows/publish-maven-snapshots.yml (hunk 4 above)
Comment (peternied) [reply .github/workflows/publish-maven-snapshots.yml:21]: FYI - I didn't make any changes to this job, its merged directly from what is in 1.3.  Going to keep this unchanged

File: .github/workflows/publish-maven-snapshots.yml (hunk 3 above)
Comment (peternied) [reply .github/workflows/publish-maven-snapshots.yml:25]: FYI - I didn't make any changes to this job, its merged directly from what is in 1.3.  Going to keep this unchanged

**Pull request merged**